packages/message-manager/src/utils.test.ts

```diff
@@ -0,0 +1,188 @@
+import * as util from './utils';
+
+describe('utils', () => {
+  it('normalizeMessageData', () => {
+    const firstNormalized = util.normalizeMessageData(
+      '879a053d4800c6354e76c7985a865d2922c82fb5b3f4577b2fe08b998954f2e0',
+    );
+    const secondNormalized = util.normalizeMessageData('somedata');
+    expect(firstNormalized).toStrictEqual(
+      '0x879a053d4800c6354e76c7985a865d2922c82fb5b3f4577b2fe08b998954f2e0',
+    );
+    expect(secondNormalized).toStrictEqual('0x736f6d6564617461');
+  });
+
+  describe('validateSignMessageData', () => {
+    it('should throw if no from address', () => {
+      expect(() =>
+        util.validateSignMessageData({
+          data: '0x879a05',
+        } as any),
```

Comment:
Also out of scope, but I think we should prefer `@ts-expect-error` for testing invalid values, over `any`. It can catch possible regressions, like if the type were to suddenly accept `{ data: string }`.

Reply:
Oh yeah, good idea!